src/pheval/post_processing/post_processing.py

```diff
@@ -57,3 +61,103 @@ def as_dict(self):
             "score": self.pheval_variant_result.score,
             "rank": self.rank,
         }
+
+
+class ResultSorter:
+    def __init__(
+        self, pheval_results: [PhEvalGeneResult] or [PhEvalVariantResult], ranking_method: str
+    ):
+        self.pheval_results = pheval_results
+        self.ranking_method = ranking_method
+
+    def sort_by_decreasing_score(self):
+        """Sort results in descending order."""
+        return sorted(self.pheval_results, key=operator.attrgetter("score"), reverse=True)
+
+    def sort_by_increasing_score(self):
+        """Sort results in ascending order."""
+        return sorted(self.pheval_results, key=operator.attrgetter("score"), reverse=False)
+
+    def sort_pheval_results(self):
+        """Sort results with best score first."""
+        return (
+            self.sort_by_increasing_score()
+            if self.ranking_method.lower() == "pvalue"
+            else self.sort_by_decreasing_score()
+        )
+
+
+@dataclass
+class ScoreRanker:
```

Comment:
add enum for ascending/descending ranking order. e.g. p-value is in ascending order 0.0 = rank 1, combined score is in descending order 1.0 = rank 1.

The score and sort/rank order should be defined in the config file - add in an explanation otherwise the scores might be ranked in the wrong order!

Reply:
tried to add in the match case statements we spoke about yesterday for throwing errors when results were not correctly sorted - however I think this is something implemented from python 3.10 onwards and so it failed when running `tox`. So instead I am using if/elif for the time being